yapapi/engine.py

```diff
@@ -556,7 +559,8 @@ async def _process_debit_notes(self) -> None:
                     agreement=agreement,
                     debit_note=debit_note,
                 )
-                await self._enforce_debit_note_intervals(job, debit_note)
+                if not await self._enforce_debit_note_intervals(job, debit_note):
```

Comment:
??? How does this "address my comments"?

If I understand correctly, this way we'll just never accept a debit note for an already terminated agreement (e.g. the last debit note). Why? What's going on here? If there is some magic logic behind this, **PLEASE** explain this in a comment.

Reply:
yes, we don't need to accept it, since, after the agreement is terminated, we'll be issued an invoice

Reply:
Fine, I understand this is more-or-less harmless, but:
1. We'll be having a growing stack of debit notes that were Received, but neither accepted or rejected
2. Provider will **also** have a growing stack of such debit notes
3. We have an event `DebitNoteReceived` that will not be followed by any event saying what happened to the DebitNote

Also, this introduces logic changes that have nothing to do with this PR & we should avoid this.

Reply:
re: 1, 2 -> I don't see the concern - debit notes are only important as long as the agreement is in action - afterwards, the payment is taken care of through the invoice and it is the invoice that matters

re: 3 - good point - if you see it as important, I'd go with a `DebitNoteIgnored` .... _or_ we can just reject them... but I think it has potential to cause more issues than just ignoring them